packages/dds/task-manager/src/taskManager.ts

```diff
@@ -312,25 +317,55 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
         return lockAcquireP;
     }
 
-    public abandon(taskId: string) {
-        if (!this.connected) {
-            throw new Error(`Attempted to abandon in disconnected state: ${taskId}`);
+    public subscribeToTask(taskId: string) {
+        if (this.subscribed(taskId)) {
+            return;
         }
 
-        // Nothing to do if we're not at least trying to get the lock.
-        if (!this.queued(taskId)) {
-            return;
+        const submitVolunteerOp = () => {
+            this.submitVolunteerOp(taskId);
+        };
+
+        const disconnectHandler = () => {
+            // Wait to be connected again and then re-submit volunteer op
+            this.connectionWatcher.once("connect", submitVolunteerOp);
+        };
+
+        const checkIfAbandoned = (eventTaskId: string) => {
+            if (eventTaskId !== taskId) {
+                return;
+            }
+
+            this.abandonWatcher.off("abandon", checkIfAbandoned);
+            this.connectionWatcher.off("disconnect", disconnectHandler);
+            this.connectionWatcher.off("connect", submitVolunteerOp);
+
+            this.subscribedTasks.set(taskId, false);
+        };
+
+        this.abandonWatcher.on("abandon", checkIfAbandoned);
+        this.connectionWatcher.on("disconnect", disconnectHandler);
+
+        if (!this.connected) {
+            disconnectHandler();
+        } else if (!this.assigned(taskId) && !this.queued(taskId)) {
+            // TODO simulate auto-ack in detached scenario
+            submitVolunteerOp();
         }
-        // TODO simulate auto-ack in detached scenario
-        if (!this.isAttached()) {
+        this.subscribedTasks.set(taskId, true);
+    }
+
+    public abandon(taskId: string) {
+        // Nothing to do if we're connected and not trying to get the lock.
+        if (this.connected && !this.queued(taskId)) {
```

Comment:
This will have us attempt to send an abandon op when disconnected, which isn't necessary -- emitting the "abandon" is important because of subscriptions, but maybe try to partition this flow so it only does the necessary steps for the state.